File: src/main/java/com/suhailkandanur/eopi/util/TreeNode.java
package com.suhailkandanur.eopi.util;

/**
 * Created by suhail on 2016-11-03.
 */
public class TreeNode {
    int data;
    TreeNode left, right, parent;

    public TreeNode(int data) {
        this.data = data;
        this.left = null;
        this.right = null;
        this.parent = null;
    }

    public int getData() {
        return data;
    }

    public TreeNode getLeft() {
        return left;
    }

    public void setLeft(TreeNode left) {
        this.left = left;
    }

    public TreeNode getRight() {
        return right;
    }

    public void setRight(TreeNode right) {
        this.right = right;
    }

    public TreeNode getParent() {
        return parent;
    }

    public void setParent(TreeNode parent) {
        this.parent = parent;
    }
}

File: python/BinarySearchTree/TreeNode.py
class TreeNode:
	def __init__(self, data, left=None, right=None):
		self.data = data
		self.left = left
		self.right = right

	def getData(self):
		return self.data

	def getLeft(self):
		return self.left

	def getRight(self):
		return self.right

	def setLeft(self, left):
		self.left = left

	def setRight(self, right):
		self.right = right
	